### ISSUE TITLE
udevd[2980]: failed to execute '/lib64/elogind/elogind-uaccess-command'

### ISSUE THREAD

**Issue by NuLL3rr0r** [fearedbliss/bliss-initramfs#31]
I am using bliss-initramfs with ZFS root and after Gentoo's full migration to elogind and pam 1.4.0 now I am getting the following:

```
[    3.590639] ZFS: Loaded module v0.8.4-r0-gentoo, ZFS pool version 5000, ZFS filesystem version 5
[    3.590894] modprobe (1970) used greatest stack depth: 13528 bytes left
[    3.592394] udevd[2177]: starting version 3.2.9
[    3.592729] random: udevd: uninitialized urandom read (16 bytes read)
[    3.592747] random: udevd: uninitialized urandom read (16 bytes read)
[    3.592774] random: udevd: uninitialized urandom read (16 bytes read)
[    3.600055] udevd[2178]: starting eudev-3.2.9
[    3.640277] udevd[2980]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/hwC1D0 ': No such file or directory
[    3.640358] udevd[2982]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/pcmC1D3p ': No such file or directory
[    3.640399] udevd[2983]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/pcmC1D8p ': No such file or directory
[    3.640909] udevd[2984]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/controlC1 ': No such file or directory
[    3.641250] udevd[2985]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/mixer1 ': No such file or directory
[    3.641344] udevd[2986]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/pcmC1D7p ': No such file or directory
[    3.650752] udevd[3009]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/audio ': No such file or directory
[    3.650770] udevd[3010]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/adsp ': No such file or directory
[    3.650834] udevd[3011]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/dsp ': No such file or directory
[    3.653259] ahci 0000:00:17.0: port does not support device sleep
[    4.174043] dmu_objset_find (3278) used greatest stack depth: 12976 bytes left
[    4.185007] zpool (3146) used greatest stack depth: 12552 bytes left
[    4.370828] z_zvol (3256) used greatest stack depth: 12376 bytes left
[    4.590433] zpool (3316) used greatest stack depth: 12128 bytes left
[    4.725885] printk: udevd: 19 output lines suppressed due to ratelimiting
[    5.611415] udevd[4288]: starting version 3.2.9
[    5.641263] udevd[4288]: starting eudev-3.2.9
```

Here is [my thread on Gentoo forums](https://forums.gentoo.org/viewtopic-p-8476174.html#8476174) describing the problem. It seems [there is a solution for Dracut](https://forums.gentoo.org/viewtopic-p-8476174.html#8476174). How can I achieve the same and add /lib64/elogind/elogind-uaccess-command to bliss-initramfs image?

**Comment by fearedbliss** [comment]
Hello,

I haven’t ran into this issue since I use systemd on Gentoo but I think easiest workaround is to try and add that file to the following list and regenerate the image. Let me know if that works.

https://github.com/fearedbliss/bliss-initramfs/blob/master/pkg/hooks/Base.py#L32

**Comment by NuLL3rr0r** [comment]
Thank you so much for the quick response. I did edit the file at /opt/bliss-initramfs/pkg/hooks/Base.py as you suggested (I am using the ebuild, so it seems after each upgrade I have to edit the file):

```
# Dependencies
    # 1. sys-apps/busybox
    # 2. sys-apps/kmod
    # 3. app-shells/bash
    # 4. sys-apps/grep
    # 5. sys-apps/kbd
    # 6. udev
    _files = [
        "/lib64/elogind/elogind-uaccess-command",
        "/sbin/modprobe",
        "/usr/sbin/crda",
        "/bin/busybox",
        Tools.GetProgramPath("kmod"),
```

And all those errors inside /var/log/dmesg are gone:

```
[    3.639833] udevd[2980]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/hwC1D0 ': No such file or directory
[    3.639882] udevd[2983]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/controlC1 ': No such file or directory
[    3.639895] udevd[2982]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/pcmC1D3p ': No such file or directory
[    3.640308] udevd[2985]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/pcmC1D8p ': No such file or directory
[    3.644150] udevd[2984]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/pcmC1D7p ': No such file or directory
[    3.648099] udevd[3001]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/adsp ': No such file or directory
[    3.648493] udevd[3005]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/snd/pcmC0D0c ': No such file or directory
[    3.648770] udevd[3002]: failed to execute '/lib64/elogind/elogind-uaccess-command' '/lib64/elogind/elogind-uaccess-command /dev/audio ': No such file or directory

[    3.676066] udevd[3054]: failed to execute '/usr/sbin/crda' '/usr/sbin/crda': No such file or directory
[    3.696545] udevd[3104]: failed to execute '/sbin/modprobe' '/sbin/modprobe zfs': No such file or directory
[    3.712374] udevd[3123]: failed to execute '/sbin/modprobe' '/sbin/modprobe zfs': No such file or directory
[    4.639328] udevd[3601]: failed to execute '/usr/sbin/crda' '/usr/sbin/crda': No such file or directory
```

It would be nice if bliss-initramfs provide a way for adding those files.

And, one more question, I also see these errors:

```
[    1.268367] Intel(R) Wireless WiFi driver for Linux
[    1.269469] Copyright(c) 2003- 2015 Intel Corporation
[    1.273626] iwlwifi 0000:03:00.0: Direct firmware load for iwlwifi-8000C-36.ucode failed with error -2
[    1.274791] iwlwifi 0000:03:00.0: Direct firmware load for iwlwifi-8000C-35.ucode failed with error -2
[    1.276050] usbcore: registered new interface driver rt2800usb
[    1.276583] iwlwifi 0000:03:00.0: loaded firmware version 34.0.1 op_mode iwlmvm
...
[    1.460789] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[    1.462943]  nvme0n1: p1 p2 p3 p4
[    1.464380] cfg80211: Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[    1.464913] platform regulatory.0: Direct firmware load for regulatory.db failed with error -2
[    1.465468] cfg80211: failed to load regulatory.db
...
[    3.580751] Bluetooth: hci0: Bootloader revision 0.0 build 2 week 52 2014
[    3.587784] Bluetooth: hci0: Device revision is 5
[    3.587785] Bluetooth: hci0: Secure boot is enabled
[    3.587786] Bluetooth: hci0: OTP lock is enabled
[    3.587786] Bluetooth: hci0: API lock is enabled
[    3.587787] Bluetooth: hci0: Debug lock is disabled
[    3.587787] Bluetooth: hci0: Minimum firmware build 1 week 10 2014
[    3.587812] bluetooth hci0: Direct firmware load for intel/ibt-11-5.sfi failed with error -2
[    3.587813] Bluetooth: hci0: Failed to load Intel firmware file (-2)
```

Is it possible to solve these issues the same way? Or, there's another way for embedding firmwares?

**Comment by fearedbliss** [comment]
Hey @NuLL3rr0r, I'm glad that adding it to the files worked.

Embedding firmware is also supported with a similar approach, check out the usage for that here:

https://github.com/fearedbliss/bliss-initramfs/blob/master/USAGE#L353

The original architecture for bliss-initramfs was to allow easy configuration of the initramfs through the existing .py files. Basically I decided (a long time ago) to use Python's file level module capabilities as an easy way to load stuff into the initramfs. Thing is that this configuration wasn't designed to be placed (let's say) in /etc. This would allow Gentoo's CONFIG_PROTECT to detect that a file in there has been modified by the user (during package install, upgrades, etc in Portage) and thus would prompt you to resolve these discrepancies. Since the project is very old at this point and it's primarily in maintenance mode, and also because of life priorities, I'm not inclined to change it. However, I made this PR 2 years ago that was made to tackle this issue specifically:

https://github.com/fearedbliss/bliss-initramfs/issues/21

I am open to letting someone else do this rearchitecting of bliss-initramfs' configuration system so that it could do this. If after reviewing it everything looks fine, I wouldn't have any objections to merging that work in.
